File: Hash Table/274_HIndex.java
public class Solution {
    public int hIndex(int[] citations) {
        int n = citations.length;
        if (n == 0 || (n == 1 && citations[0] == 0)) return 0;
        int hIndexVal = 0;
        for (int i = 1; i <= n; i++) {
            int bigger = 0, smaller = 0, equals = 0;
            boolean isHIndex = true;
            for (int j = 0; j < n; j++) {
                int val = citations[j];
                if (val > i) {
                    bigger++;
                } else if (val < i) {
                    smaller++;
                } else {
                    equals++;
                }
                if (bigger > i || smaller > n - i) {
                    isHIndex = false;
                    break;
                }
            }
            if (isHIndex) {
                hIndexVal = Math.max(hIndexVal, i);
            }
        }
        return hIndexVal;
    }
}

File: 274_H-Index.py
class Solution(object):
    def hIndex(self, citations):
        """
        :type citations: List[int]
        :rtype: int
        """
        n = len(citations)
        if (n == 0):
            return 0
        sums = [0] * (n + 1)

        for cit in citations:
            if cit == 0:
                continue
            if cit >= n:
                sums[n] += 1
            else:
                sums[cit] += 1

        for i in range(n, -1, -1):
            if i < n:
                sums[i] += sums[i + 1]
            if sums[i] >= i:
                return i
        return 0